server/router/rider.ts

```diff
@@ -80,6 +80,26 @@ router.get('/:id/currentride', validateUser('Rider'), (req, res) => {
   });
 });
 
+router.get('/:id/usage', validateUser('User'), (req, res) => {
+  const { params: { id } } = req;
+  let noShowCount: number;
+  let studentRides: number;
+  db.getById(res, Rider, id, tableName, () => {
+    const isRider = new Condition('rider').eq(id);
+    const isNoshow = new Condition('status').eq(Status.NO_SHOW);
+    const conditionNoShow = isRider.group(isNoshow);
+    const isCompleted = new Condition('status').eq(Status.COMPLETED);
+    const conditionRides = isRider.group(isCompleted);
+    db.scan(res, Ride, conditionNoShow, (data: RideType[]) => {
+      noShowCount = data.length;
+      db.scan(res, Ride, conditionRides, (data_: RiderType[]) => {
+        studentRides = data_.length;
+        res.send({ studentRides, noShowCount });
+      });
+    });
```

Comment:
to reduce database reads, you could use the isRider condition, then do two .filter() calls, one for no show, one for completed, on the data sent from the callback